mParticle-Apple-SDK/mParticle.m

```diff
@@ -163,30 +163,47 @@ @implementation MPRokt
 
 - (void)selectPlacements:(NSString *)identifier
               attributes:(NSDictionary<NSString *, NSString *> * _Nullable)attributes {
-    for (NSString *key in attributes) {
-        [[MParticle sharedInstance].identity.currentUser setUserAttribute:key value:attributes[key]];
-    }
-    
-    dispatch_async(dispatch_get_main_queue(), ^{
-        // Forwarding call to kits
-        MPForwardQueueParameters *queueParameters = [[MPForwardQueueParameters alloc] init];
-        [queueParameters addParameter:identifier];
-        [queueParameters addParameter:attributes];
-        [queueParameters addParameter:nil];
-        [queueParameters addParameter:nil];
-        [queueParameters addParameter:nil];
-        [queueParameters addParameter:nil];
-        [queueParameters addParameter:nil];
-        [queueParameters addParameter:nil];
+    NSArray<NSDictionary<NSString *, NSString *> *> *attributeMap = [self getRoktPlacementAttributes];
+    
+    // If attributeMap is nil the kit hasn't been initialized
+    if (attributeMap) {
+        NSMutableDictionary *mappedAttributes = attributes.mutableCopy;
+        for (NSDictionary<NSString *, NSString *> *map in attributeMap) {
+            NSString *mapFrom = map[@"map"];
+            NSString *mapTo = map[@"value"];
+            if (mappedAttributes[mapFrom]) {
+                NSString * value = mappedAttributes[mapFrom];
+                [mappedAttributes removeObjectForKey:mapFrom];
+                mappedAttributes[mapTo] = value;
+            }
+        }
+        for (NSString *key in mappedAttributes) {
+            [[MParticle sharedInstance].identity.currentUser setUserAttribute:key value:mappedAttributes[key]];
+        }
```

Comment:
can this be DRYed up into a separate function so that it's called from both `selectPlacements` methods?